libvips/convolution/im_aconv.c

```diff
@@ -1005,9 +1005,7 @@ aconv_horizontal( Boxes *boxes, IMAGE *in, IMAGE *out )
 
 #define CLIP_UCHAR( V ) \
 G_STMT_START { \
-	if( (V) < 0 ) \
-		(V) = 0; \
-	else if( (V) > UCHAR_MAX ) \
+	if( (V) > UCHAR_MAX ) \
```

Comment:
Well checking an unsigned int for < 0 is a nop. If you subtract 1 from a uint32 you get 2^32-1 so it'll never be smaller than 0. If you wanted to use that code as is you'd have to use signed ints for yor sum arg to VCONV.